File: ecom-backend/src/main/java/com/example/demo/Controller/WishlistController.java
package com.example.demo.Controller;

import com.example.demo.Model.Wishlist;
import com.example.demo.Response.CustomResponse;
import com.example.demo.Service.WishlistService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/wishlist")
@CrossOrigin("*")
public class WishlistController {
    @Autowired
    WishlistService wishlistService;

    @PostMapping("/add/{userId}/{productId}")
    public ResponseEntity<?> addWishlistItem(@PathVariable String userId, @PathVariable String productId) {
        wishlistService.addWishlistItem(userId, productId);
        return ResponseEntity.status(HttpStatus.CREATED).body(new CustomResponse(HttpStatus.CREATED.value(), "created", null));
    }

    @GetMapping("/all/{userId}")
    public ResponseEntity<?> getALLWishlistItems(@PathVariable String userId){
        return ResponseEntity.status(HttpStatus.OK).body(new CustomResponse(HttpStatus.OK.value(), "success", wishlistService.getUserWishlist(userId)));
    }

    @DeleteMapping("/del/{wishlistId}")
    public ResponseEntity<?> deleteWishlistItem(@PathVariable String wishlistId){
        wishlistService.removeFromWishlist(wishlistId);
        return ResponseEntity.status(HttpStatus.OK).body(new CustomResponse(HttpStatus.OK.value(), "deleted", null));
    }


}

File: bank/src/main/kotlin/com/junocoding/bank/controllers/wishlistController.kt
package com.junocoding.bank.controllers

import com.junocoding.bank.models.Product
import com.junocoding.bank.models.wishList
import com.junocoding.bank.repositories.productRepository
import com.junocoding.bank.repositories.wishListRepository
import com.junocoding.bank.service.wishlistService
import org.springframework.web.bind.annotation.CrossOrigin
import org.springframework.web.bind.annotation.DeleteMapping
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@CrossOrigin("*")
@RestController
@RequestMapping("/products/wishlist")
class wishlistController(private val wishListRepository: wishListRepository,
                         private val productRepository: productRepository,
    private val wishlistService: wishlistService
    ){
    //adding item to wishlist
    @PostMapping("/add/{userId}/{id}")
    fun addtoWishlist(@PathVariable id:Long,@PathVariable userId:Long): Any/*ResponseEntity<String>*/{
        val product=productRepository.findById(id)
            .orElseThrow { NoSuchElementException("product not found") }
        val exiteditem=wishListRepository.findByUserId(userId).find { it.product.id==id }
        if(exiteditem!=null){
            //return ResponseEntity.badRequest().body("Product already exists")
            return "Product already exists"
        }
        val newItem=wishList(product =product, userId = userId)
        return wishListRepository.save(newItem)
        //return ResponseEntity.ok("product added to wishlist")
    }
    //removing item form wishlist
    @DeleteMapping("/del/{id}")
    fun deleteWishlistItem(@PathVariable id:Long){
        return wishlistService.deleteWishlistitem(id)
    }
    //getting all wishlist items
    @GetMapping("/items/{userId}")
    fun getAllWishlistItems(@PathVariable userId: Long): List<wishList> {
        return wishlistService.getallItemsbyuserwishlist(userId)
    }

}